springwolf-add-ons/springwolf-kotlinx-serialization-model-converter/src/main/kotlin/io/github/springwolf/addons/kotlinx_serialization_model_converter/converter/KotlinxSerializationModelConverter.kt

```diff
@@ -0,0 +1,143 @@
+import kotlin.reflect.full.memberProperties
+import kotlin.reflect.jvm.javaType
+import kotlin.reflect.jvm.jvmErasure
+
+class KotlinxSerializationModelConverter(private val useFqn: Boolean = false) : ModelConverter {
+
+    private val qualifiedNameMap: MutableMap<String, String> = ConcurrentHashMap()
+
+    override fun resolve(
+        annotatedType: AnnotatedType, context: ModelConverterContext, chain: Iterator<ModelConverter>
+    ): Schema<*>? {
+        if (annotatedType.type is Class<*>) {
+            val kClass = (annotatedType.type as Class<*>).kotlin
+            val isKotlinSerializable = kClass.findAnnotation<Serializable>() != null
+
+            if (isKotlinSerializable) {
+                val schema = ObjectSchema()
+                schema.nullable = false
+                schema.name = kClass.simpleName
+
+                kClass.memberProperties.forEach { property ->
+                    val propertySchema = getPropertySchema(property, context)
+
+                    schema.addProperty(propertySchema.name, propertySchema)
+                    if (!propertySchema.nullable) {
+                        schema.addRequiredItem(propertySchema.name)
+                    }
+                }
+
+                return schema
+            }
+        }
+
+        return if ((chain.hasNext())) chain.next().resolve(annotatedType, context, chain) else null
+    }
+
+    private fun getPropertyName(property: KProperty1<*, *>): String {
+        val serialNameAnnotation = property.findAnnotation<SerialName>()
+        if (serialNameAnnotation != null) {
+            return serialNameAnnotation.value
+        }
+        return property.name
+    }
+
+    private fun getPropertySchema(property: KProperty1<*,*>, context: ModelConverterContext) : Schema<*> {
+        val propertySchema: Schema<*>
+
+        val name = getPropertyName(property)
+
+        if (property.returnType.jvmErasure.java.isAssignableFrom(List::class.java)) {
+            propertySchema = ArraySchema()
+            val value = (property.returnType.javaType as ParameterizedType).actualTypeArguments[0]
+            propertySchema.items = resolveRefSchema(value, context)
+        } else if (property.returnType.jvmErasure.java.isAssignableFrom(Set::class.java)) {
+            propertySchema = ArraySchema()
+            val value = (property.returnType.javaType as ParameterizedType).actualTypeArguments[0]
+            propertySchema.items = resolveRefSchema(value, context)
+            propertySchema.uniqueItems = true
+        } else if (property.returnType.jvmErasure.java.isAssignableFrom(Map::class.java)) {
+            val mapType = (property.returnType.javaType as ParameterizedType)
+            val value = mapType.actualTypeArguments[1]
+            propertySchema = MapSchema().additionalProperties(context.resolve(AnnotatedType(value)))
+        } else if (property.returnType.jvmErasure.java.isAssignableFrom(Byte::class.java)) {
+            propertySchema = IntegerSchema()
```

Comment:
Why do we need to convert Byte to Integerschema? https://www.asyncapi.com/docs/reference/specification/v3.0.0#dataTypeFormat looks like StringSchema with format byte would be the correct schema type and its also what would be generated by default...

Reply:
@sam0r040 good catch. It's now fixed